internal/core/services/events_forwarder/events_forwarder_service.go

```diff
@@ -24,60 +24,169 @@ package events_forwarder
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"strconv"
+
+	"github.com/topfreegames/maestro/internal/core/entities"
+	"github.com/topfreegames/maestro/internal/core/entities/game_room"
+
+	"github.com/topfreegames/maestro/internal/core/entities/forwarder"
 
 	"github.com/topfreegames/maestro/internal/core/entities/events"
 
-	"github.com/topfreegames/maestro/internal/core/entities/game_room"
 	"github.com/topfreegames/maestro/internal/core/ports"
 	"github.com/topfreegames/maestro/internal/core/services/interfaces"
 	"go.uber.org/zap"
 )
 
 type EventsForwarderService struct {
-	eventsForwarder ports.EventsForwarder
-	logger          *zap.Logger
+	eventsForwarder  ports.EventsForwarder
+	logger           *zap.Logger
+	schedulerStorage ports.SchedulerStorage
+	instanceStorage  ports.GameRoomInstanceStorage
+	schedulerCache   ports.SchedulerCache
 }
 
-func NewEventsForwarderService(eventsForwarder ports.EventsForwarder) interfaces.EventsService {
+func NewEventsForwarderService(
+	eventsForwarder ports.EventsForwarder,
+	schedulerStorage ports.SchedulerStorage,
+	instanceStorage ports.GameRoomInstanceStorage,
+	schedulerCache ports.SchedulerCache,
+) interfaces.EventsService {
 	return &EventsForwarderService{
 		eventsForwarder,
 		zap.L().With(zap.String("service", "rooms_api")),
+		schedulerStorage,
+		instanceStorage,
+		schedulerCache,
 	}
 }
 
 func (es *EventsForwarderService) ProduceEvent(ctx context.Context, event *events.Event) error {
-	room := &game_room.GameRoom{}
-	instance := &game_room.Instance{}
-
-	switch event.Name {
-	case events.RoomEvent:
-		return es.forwardRoomEvent(ctx, room, instance, map[string]interface{}{}, "")
-	case events.PlayerEvent:
-		return es.forwardPlayerEvent(ctx, room, map[string]interface{}{}, "")
+	var eventType string
```

Comment:
To be more readable, you can switch if condition to avoid else statement, like this: 

if _, ok := event.Attributes["eventType"]; !ok { 
	return errors.New("eventAttributes must contain key \"eventType\"")
}

eventType := event.Attributes["eventType"].(string)
…..

Reply:
Nice!!